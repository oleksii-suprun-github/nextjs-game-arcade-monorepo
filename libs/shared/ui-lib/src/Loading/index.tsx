function Loading({
  fullscreen,
  children,
  customClasses,
}: {
  fullscreen?: boolean;
  children?: React.ReactNode;
  customClasses?: string;
}) {
  return (
    <div>
      {fullscreen ? (
        <div
          data-testid="loading-ui-component"
          className="fixed inset-0 z-50 flex items-center justify-center bg-gray-800 bg-opacity-75"
        >
          <span className="loading loading-ring loading-lg"></span>
        </div>
      ) : (
        <div data-testid="loading-ui-component" className={`loading loading-lg ${customClasses}`}>
          {children}
        </div>
      )}
    </div>
  );
}
export default Loading;
