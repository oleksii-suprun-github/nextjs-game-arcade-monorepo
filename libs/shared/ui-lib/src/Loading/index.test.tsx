import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import Loading from '.';

describe('Loading', () => {
  it('renders correctly with children', () => {
    const { container } = render(<Loading>Loading...</Loading>);
    expect(container).toMatchSnapshot();
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });

  it('applies the correct class names', () => {
    render(<Loading>Loading...</Loading>);
    const divElement = screen.getByTestId('loading-ui-component');
    expect(divElement).toHaveClass('loading');
    expect(divElement).toHaveClass('loading-lg');
  });

  it('applies custom classes correctly', () => {
    const customClasses = 'custom-class another-class';
    render(<Loading customClasses={customClasses}>Loading...</Loading>);
    const divElement = screen.getByTestId('loading-ui-component');
    expect(divElement).toHaveClass('custom-class');
    expect(divElement).toHaveClass('another-class');
  });

  it('renders correctly without children', () => {
    render(<Loading />);
    const divElement = screen.getByTestId('loading-ui-component');
    expect(divElement).toBeInTheDocument();
    expect(divElement).toBeEmptyDOMElement();
  });

  it('renders correctly fullscreen loader', () => {
    render(<Loading fullscreen />);
    const divElement = screen.getByTestId('loading-ui-component');
    expect(divElement).toBeInTheDocument();
    expect(divElement).not.toBeEmptyDOMElement();
  });
});
