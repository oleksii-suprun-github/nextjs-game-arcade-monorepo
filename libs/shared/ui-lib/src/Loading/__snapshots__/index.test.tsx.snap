// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading > renders correctly with children 1`] = `
<div>
  <div
    class="loading loading-lg undefined"
    data-testid="loading-ui-component"
  >
    Loading...
  </div>
</div>
`;
